File: viewModelDemo/app/src/main/java/edu/cs4730/viewmodeldemo/DataViewModel.java
package edu.cs4730.viewmodeldemo;

import androidx.lifecycle.ViewModel;

/**
 * this is a simple viewmodel class holding two data items.
 */

public class DataViewModel extends ViewModel {

    int count = 0;
    String name = "";

    String getCount() {
        return String.valueOf(count);
    }

    void incrementCount() {
        count++;
    }
}

File: viewModelDemo_kt/app/src/main/java/edu/cs4730/viewmodeldemo_kt/DataViewModel.kt
package edu.cs4730.viewmodeldemo_kt

import androidx.lifecycle.ViewModel

/**
 * this is a simple viewmodel class holding two data items.
 */
class DataViewModel : ViewModel() {
    var count = 0
    var name = ""
    fun getCount(): String {
        return count.toString()
    }

    fun incrementCount() {
        count++
    }
}